app/controllers/registration_controller.rb

```diff
@@ -274,6 +274,26 @@ def import
 
   private
 
+    def action_type(request)
+      self_updating = request[:user_id] == @current_user
+      status = request.dig('competing', 'status')
+      if status == 'cancelled'
+        return self_updating ? 'Competitor delete' : 'Admin delete'
+      end
+      self_updating ? 'Competitor update' : 'Admin update'
```

Comment:
For future-proofness (especially when we get automated acceptance which means more action_types to take into account) you should probably translate these to integer codes and have i18n translations for the codes, similar to how we already handle some error messages.

Reply:
(feel free to push this to a follow-up PR if you feel that it's too much work)

Reply:
I'll do it in another PR because it also requires new I18n strings